lib/gemstash/gem_pusher.rb

```diff
@@ -38,12 +39,27 @@ def storage
       @storage ||= Gemstash::Storage.for("private").for("gems")
     end
 
+    def spec_storage
+      @spec_storage ||= Gemstash::Storage.for("private").for("specs")
+    end
+
+    def full_name
+      @full_name ||= gem.spec.full_name
+    end
+
     def check_auth
       Gemstash::Authorization.check(@auth_key, "push")
     end
 
     def store_gem
-      storage.resource(gem.spec.full_name).save(@content, indexed: true)
+      storage.resource(full_name).save(@content, indexed: true)
+    end
+
+    def store_gemspec
+      spec = gem.spec
+      spec = Marshal.dump(spec)
+      spec = Zlib::Deflate.deflate(spec)
+      spec_storage.resource(full_name).save(spec)
     end
 
```

Comment:
Not quite on this particular block of code. This block is about storing the `Gem::Specification` instance so it can be retrieved when `/quick/Marshal.4.8/gemname-0.1.0.gemspec.rz` is called. There is a different class that I am using that *is* similar concern to yours, and that would be `Gemstash::SpecsBuilder` which I put in `lib/gemstash/secs_builder.rb`. The difference is that the specs.4.8.gz is a Marshalled and Gzipped array of arrays, where each array has [String<name>, Gem::Version, String<platform>], this is storing a Marshalled Zlib::Deflated instance of the Gem::Specification.